tritonbench/components/ncu/nsys_analyzer.py

```diff
@@ -0,0 +1,95 @@
+import csv
+import os
+import shutil
+import sys
+from typing import Dict, List
+
+# The nsys metrics to the reports. The value is the list of reports of nsys.
+nsys_metrics_to_reports = {
+    # the sum of kernel execution time
+    "nsys_gpu_kernel_sum": ["cuda_gpu_kern_sum", "nvtx_sum"],
+    # the overhead of kernel launch
+    "nsys_launch_overhead": ["cuda_gpu_kern_sum", "nvtx_sum"],
+    # the names of kernels
+    "nsys_kernel_names": ["cuda_gpu_kern_sum"],
+    # the durations of kernels
+    "nsys_kernel_durations": ["cuda_gpu_kern_sum"],
+    # the duration of nvtx range
+    "nsys_nvtx_range_duration": ["nvtx_sum"],
+    # the number of kernels
+    "nsys_num_of_kernels": ["cuda_gpu_kern_sum"],
+}
+# The public nsys metrics to tritonbench
+nsys_bench_metrics = list(nsys_metrics_to_reports.keys())
+
+
+def read_nsys_report(
+    report_path: str, required_metrics: List[str]
+) -> Dict[str, List[float]]:
+    assert os.path.exists(
+        report_path
+    ), f"The nsys report at {report_path} does not exist. Ensure you add --metrics nsys_rep to your benchmark run."
+    reports_required = []
+    for metric in required_metrics:
+        if metric in nsys_metrics_to_reports:
+            reports_required.extend(nsys_metrics_to_reports[metric])
+    reports_required = list(set(reports_required))
+    assert reports_required, "No nsys reports required"
+    cmd = f"nsys stats --report {','.join(reports_required)} --force-export=true --format csv --output . --force-overwrite=true {report_path} > /dev/null 2>&1"
```

Comment:
There are multiple ways to obtain reports and collect data from them. 1. run nsys with desired reports via `--report` . It will automatically generate the csv files for each report. 2. only generate sqlite and utilize report scripts in ` nsys/target-linux-x64/reports`. See [discussion here](https://forums.developer.nvidia.com/t/how-to-use-the-report-scripts/313987). In summary, we always need one system command call for nsys to either generate sqlite or csv reports, and the report scripts in nsys folder also does csv process, so we can directly utilize nsys to obtain csv report files rather than processing by ourselves.